modules/postgres/postgres.go

```diff
@@ -217,6 +217,7 @@ func (c *PostgresContainer) Snapshot(ctx context.Context, opts ...SnapshotOption
 	// execute the commands to create the snapshot, in order
 	if err := c.execCommandsSQL(ctx,
 		// Drop the snapshot database if it already exists
+		fmt.Sprintf(`UPDATE pg_database SET datistemplate = FALSE WHERE datname = '%s'`, snapshotName),
```

Comment:
Nit: Can we refer to the source for this, maybe in a comment, something like https://stackoverflow.com/questions/11388786/how-to-drop-a-template-database-from-postgresql, or the original docs?